### PR TITLE
Fix array_transform to not recompute the argument

### PR DESCRIPTION
**Pull request by revans2**
This fixes https://github.com/NVIDIA/spark-rapids/issues/9472

Thanks to @razajafri for doing the majority of the debugging on this.

### CODE REVIEW

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Review by razajafri — APPROVED** [review]
LGTM. other than a nit

File: sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java [hunk 1]
```diff
@@ -895,6 +895,22 @@ public static ColumnarBatch combineColumns(ColumnarBatch ... batches) {
     return incRefCounts(ret);
   }
 
+  public static ColumnarBatch combineColumns(ColumnarBatch cb, GpuColumnVector ... vectors) {
+    final int numRows = cb.numRows();
+    ArrayList<ColumnVector> columns = new ArrayList<>();
```
Comment (jlowe) [inline sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java:900]: Nit: We know exactly how big the array needs to be and can hint this here.
```suggestion
    ArrayList<ColumnVector> columns = new ArrayList<>(cb.numCols() + vectors.length);
```

**Review by jlowe — DISMISSED** [review]

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@razajafri and @jlowe please take another look

**Review by jlowe — APPROVED** [review]

**Review by razajafri — APPROVED** [review]

**Pull request merged**